src/cogent3/core/sequence.py

```diff
@@ -1942,6 +1942,8 @@ def __init__(
         self.step = step
         self._offset = offset
         self._seqid = seqid
+        if seq_len and seq_len != len(seq):
```

Comment:
this should still be `is None` because `seq_len=0` and a `seq="AGT"` would not be caught by this check